lib/fileHandler/index.js

```diff
@@ -324,42 +335,68 @@ module.exports = function (options) {
 		};
 	}
 
+	function checkExpiryConditions (url, headers, rawBody) {
+		return (
+			_.includes(url, '.amazonaws.com/') &&
+			_.includes(headers['content-type'], 'application/xml') &&
+			_.includes(headers.server, 'AmazonS3') &&
+			_.includes(rawBody.toString(), 'Request has expired')
+		);
+	}
+
+	function expiryRejectObject () {
+		return {
+			code: '#user_input_error',
+			message: 'The file provided has expired.',
```

Comment:
We can probably make this more helpful to users, would be nice to add something about links expiring after 6 hours. If we looked at query params we can get the exact time it expires, so we can tell our customers when it actually expired if we want aswell.